sbi/analysis/plot.py

```diff
@@ -976,3 +978,353 @@ def _get_default_opts():
+                        xlabel=f"posterior rank {parameter_labels[jj]}",
+                        # Show legend and ylabel only in first subplot.
+                        show_ylabel=jj == 0,
+                        show_legend=jj == 0,
+                        alpha=line_alpha,
+                    )
+                    if ii == 0 and show_uniform_region:
+                        plot_cdf_region_expected_under_uniformity(
+                            num_sbc_runs, num_bins, num_repeats, alpha=0.1
+                        )
+                elif plot_type == "hist":
+                    plot_ranks_as_hist(
+                        ranki[:, jj],
+                        num_bins,
+                        num_posterior_samples,
+                        ranks_label=ranks_labels[ii],
+                        color="firebrick" if colors is None else colors[ii],
+                        xlabel=f"posterior rank {parameter_labels[jj]}",
+                        # Show legend and ylabel only in first subplot.
+                        show_ylabel=show_ylabel,
+                        show_legend=jj == 0,
+                        alpha=line_alpha,
+                        xlim_offset_factor=xlim_offset_factor,
+                    )
+                    # Plot expected uniform band.
+                    plot_hist_region_expected_under_uniformity(
+                        num_sbc_runs,
+                        num_bins,
+                        num_posterior_samples,
+                        alpha=uniform_region_alpha,
+                    )
+                else:
+                    raise ValueError(
+                        f"plot_type {plot_type} not defined, use one in {plot_types}"
+                    )
+                # Remove empty subplots.
+        col_idx += 1
+        while num_rows > 1 and col_idx < num_cols:
+            ax[row_idx, col_idx].axis("off")
+            col_idx += 1
+
+    # When there is only one set of ranks show all params in a single subplot.
+    else:
+        if fig is None or ax is None:
+            fig, ax = plt.subplots(1, 1, figsize=figsize)
+
+        plt.sca(ax)
+        ranki = ranks[0]
+        for jj in range(num_parameters):
+            plot_ranks_as_cdf(
+                ranki[:, jj],
+                num_bins,
+                num_repeats,
+                ranks_label=parameter_labels[jj],
+                color=f"C{jj}" if colors is None else colors[jj],
+                xlabel=f"posterior rank",
+                # Plot ylabel and legend at last.
+                show_ylabel=jj == (num_parameters - 1),
+                show_legend=jj == (num_parameters - 1),
+                alpha=line_alpha,
+            )
+        if show_uniform_region:
+            plot_cdf_region_expected_under_uniformity(
+                num_sbc_runs, num_bins, num_repeats, alpha=uniform_region_alpha
+            )
+
+    return fig, ax
+
+
+def plot_ranks_as_hist(
```

Comment:
Is this user facing or should it be hidden `_plot_ranks_as_hist()`?

Same for the methods below